core/cfg/ini.cpp

```diff
@@ -1,299 +1,280 @@
+		return default_value;
+	}
+	else
+	{
+		return entry->get_bool();
 	}
-	return 0;
 }
 
-ConfigSection* ConfigFile::GetEntry(string name)
+void ConfigFile::set(string section_name, string entry_name, string value, bool is_virtual)
 {
-	ConfigSection* c=FindSection(name);
-	if (!c)
+	ConfigSection* section = this->get_section(section_name, is_virtual);
+	if(section == NULL)
 	{
-		entrys=c= new ConfigSection(entrys);
-		c->name=name;
+		section = this->add_section(section_name, is_virtual);
 	}
+	section->set(entry_name, value);
+};
 
-	return c;
+void ConfigFile::set_int(string section_name, string entry_name, int value, bool is_virtual)
+{
+	std::stringstream str_value;
+	str_value << value;
+	this->set(section_name, entry_name, str_value.str(), is_virtual);
 }
 
-ConfigFile::~ConfigFile()
+void ConfigFile::set_bool(string section_name, string entry_name, bool value, bool is_virtual)
 {
-	ConfigSection* n=entrys;
-	
-	while(n)
-	{
-		ConfigSection* p=n; 
-		n=n->next;
-		delete p;
-	}
+	string str_value = (value ? "yes" : "no");
+	this->set(section_name, entry_name, str_value, is_virtual);
 }
 
-void ConfigFile::ParseFile(FILE* file)
+void ConfigFile::parse(FILE* file)
 {
-	wchar line[512];
-	wchar cur_sect[512]={0};
-	int cline=0;
+	if(file == NULL)
+	{
+		return;
+	}
+	char line[512];
+	char current_section[512] = { '\0' };
+	int cline = 0;
 	while(file && !feof(file))
 	{
-		fgets(line,512,file);
+		fgets(line, 512, file);
 		if (feof(file))
+		{
 			break;
+		}
 
 		cline++;
 
-		if (strlen(line)<3)
+		if (strlen(line) < 3)
+		{
 			continue;
-		if (line[strlen(line)-1]=='\r' || line[strlen(line)-1]=='\n')
-			line[strlen(line)-1]=0;
+		}
+
+		if (line[strlen(line)-1] == '\r' ||
+			  line[strlen(line)-1] == '\n')
+		{
+			line[strlen(line)-1] = '\0';
+		}
+
+		char* tl = trim_ws(line);
 
-		wchar* tl=trim_ws(line);
-		if (tl[0]=='[' && tl[strlen(tl)-1]==']')
+		if (tl[0] == '[' && tl[strlen(tl)-1] == ']')
 		{
-			tl[strlen(tl)-1]=0;
-			strcpy(cur_sect,tl+1);
-			trim_ws(cur_sect);
+			tl[strlen(tl)-1] = '\0';
+			strcpy(current_section, tl+1);
+			trim_ws(current_section);
 		}
 		else
 		{
-			if (cur_sect[0]==0)
-				continue;//no open section
-			wchar* str1=strstr(tl,"=");
-			if (!str1)
+			if (strlen(current_section) == 0)
+			{
+				continue; //no open section
+			}
+
+			char* separator = strstr(tl, "=");
+
+			if (!separator)
 			{
-				printf("Malformed entry on config - ignoring @ %d(%s)\n",cline,tl);
+				printf("Malformed entry on config - ignoring @ %d(%s)\n",cline, tl);
 				continue;
 			}
-			*str1=0;
-			str1++;
-			wchar* v=trim_ws(str1);
-			wchar* k=trim_ws(tl);
-			if (v && k)
+
+			*separator = '\0';
+
+			char* name = trim_ws(tl);
+			char* value = trim_ws(separator + 1);
+			if (name == NULL || value == NULL)
 			{
-				ConfigSection*cs=this->GetEntry(cur_sect);
-				
-				//if (!cs->FindEntry(k))
-				cs->SetEntry(k,v,CEM_SAVE|CEM_LOAD);
+				printf("Malformed entry on config - ignoring @ %d(%s)\n",cline, tl);
```

Comment:
An mepty size if recognised as an error. This is consistent with the previous behaviour but I wonder if this what we want especially since there is this code:

~~~c
settings.reios.ElfFile = cfgLoadStr("reios", "ElfFile","");
~~~

which initialised this as an empty string by default.